### DISCUSSION TITLE
qmake symlink autmatically removed?

### DISCUSSION THREAD

**Discussion by gernophil** [Homebrew/discussions#4771] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: 12c8778af976269344f3c0b914243e178fc4ffbd
Last commit: 18 hours ago
Core tap JSON: 07 Sep 19:43 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.TSv4aSEba9/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libomp.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/omp-tools.h
  /usr/local/include/omp.h
  /usr/local/include/ompt.h
```

### Description of issue

Hey everyone,
I just installed `gt5` with `brew install qt5`. After that `qmake` (and `which qmake`) did immediately work. Then I did an `brew update` and `brew upgrade` and left my Mac. After I returned `qmake` (and `which qmake`) just showed a command not found error. The symlink is just gone and after reinstalling (with `reinstall` and with `uninstall` followed by `install`) I now get the message:
```
If you need to have qt@5 first in your PATH, run:
  echo 'export PATH="/opt/homebrew/opt/qt@5/bin:$PATH"' >> ~/.zshrc
```
That explains, why `which qmake` just return a command not found. However, can an `brew update && brew upgrade` lead to a loss of a symlink? This feels a bit weird to me.

**Comment by SMillerDev** [comment]
If your old install predates qt6 it could because then qt5 was the main qt

**Reply by gernophil** [reply]
The old install was just some hours ago (yesterday morning with `brew install qt5` (not `brew install qt@5`) :).

**Reply by gromgit** [reply]
`qt5` is an alias for the `qt@5` formula, which is _keg-only_ (i.e. its binaries are _not_ found in a standard Homebrew-augmented PATH). Therefore, the `qmake` you ran before as simply `qmake` was not from Homebrew's `qt@5` formula. To get that, you either have to specify the full path to it (`/opt/homebrew/opt/qt@5/bin/qmake`) or add the `qt@5` bin directory to your PATH as instructed by the message you posted.

**Reply by gernophil** [reply]
It has to have been Homebrew `qmake`, since I don't have (and also did not have) any other `qmake` installed. That's why I am wondering.

Before the install with brew `which qmake` resulted in `qmake not found`. After installing it showed Homebrews path and after upgrading it was again `qmake not found`.

**Reply by gromgit** [reply]
> After installing it showed Homebrews path

What exactly do you mean by this? What were the command(s) you ran that "showed Homebrew's path", and what exactly did they output?

**Reply by gernophil** [reply]
"It" is the command from the sentence before (`which qmake`). And the output I don't remember completely. I also cannot reproduce this anymore (due to obvious reasons). But it contained `/opt/homebrew`.

**Reply by gromgit** [reply]
Something doesn't add up. What's the output of:
```
echo $PATH
ls -l /opt/homebrew/bin | grep qt
```

**Reply by gernophil** [reply]
That depends on my configuration of course. By default I don't have the brew path and qmake path added. If I need them I add them via a function in my .zshrc:
```
# brew
alias arm64_brew='eval "$(/opt/homebrew/bin/brew shellenv)"'
alias x86_64_brew='eval "$(/usr/local/bin/brew shellenv)"'
# qmake
arm64_qmake_path() {
    export PATH="/opt/homebrew/opt/qt@5/bin:$PATH"
}
x86_64_qmake_path() {
    export PATH="/usr/local/opt/qt@5/bin:$PATH"
}
```
After `arm64_brew && arm64_qmake_path` the output is as follows:
```
% echo $PATH
/opt/homebrew/opt/qt@5/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Apple/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin
```
```
% ls -l /opt/homebrew/bin | grep qt
lrwxr-xr-x  1 gernophil  admin    39  4 Sep 15:23 qt-faststart -> ../Cellar/ffmpeg/6.0_1/bin/qt-faststart
```
But as I already mentioned, everything works. I an just wondering, why it stopped working all of a sudden.
